deep_qa/data/instances/instance.py

```diff
@@ -334,14 +350,20 @@ def pad_sequence_to_length(sequence: List,
         much of the question set up. If you want to truncate from the other
         direction, you can.
         """
-
-        padded_sequence = []
-        for _ in range(desired_length):
-            padded_sequence.append(default_value())
-        sequence_length = min(len(sequence), desired_length)
-        if sequence_length != 0:
+        if truncate_from_right:
+            truncated = sequence[-desired_length:]
+        else:
+            truncated = sequence[:desired_length]
+        if len(truncated) < desired_length:
+            # If the length of the truncated sequence is less than the desired
+            # length, we need to pad.
+            padding_sequence = [default_value()] * (desired_length - len(truncated))
```

Comment:
I guess, now having looked at your character changes, but it should matter for background sentences, I think...